packages/core/src/math/isqrt.rs

```diff
@@ -23,9 +25,13 @@ where
     /// Algorithm adapted from
     /// [Wikipedia](https://en.wikipedia.org/wiki/Integer_square_root#Example_implementation_in_C).
     fn isqrt(self) -> Self {
-        let mut x0 = self >> 1;
+        let zero = Self::from(0);
+        if self == zero {
```

Comment:
This could be a quick check for `self <= 1`, since $\sqrt{1} = 1$

Reply:
Wikipedia agrees

```c
	// Zero yields zero
	// One yields one
	if (s <= 1) 
		return s;
```